lib/notifications/client/speaker.rb

```diff
@@ -49,15 +49,11 @@ def post(kind, form_data)
       # @param id [String]
       # @param options [Hash] query
       # @see #perform_request!
-      def get(id = nil, options = nil)
-        if !options.nil? && options.keys.any?
-          path = "?" + URI.encode_www_form(options)
-        end
-
-        request = Net::HTTP::Get.new(
-          "#{BASE_PATH}/#{id}#{path}",
-          headers
-        )
+      def get(id = nil, options = {})
+        path = BASE_PATH.dup
+        path << "/" << id if id
+        path << "?" << URI.encode_www_form(options) if options.any?
+        request = Net::HTTP::Get.new(path, headers)
         perform_request!(request)
       end
```

Comment:
Building URLs is quite a common thing to do in Ruby applications that the standard library provides some help. For example: http://stackoverflow.com/questions/6152231/is-there-a-ruby-library-gem-that-will-generate-a-url-based-on-a-set-of-parameter

It might be worth considering using something built in rather than rolling this yourself.